modules/contacts/client/components/RemoveContact.js

```diff
@@ -3,7 +3,15 @@ import PropTypes from 'prop-types';
 import { withTranslation } from 'react-i18next';
 import { Modal } from 'react-bootstrap';
 
-export function RemoveContact({ t, contact, show, inProgress, onHide, onRemoveContact, situation='confirmed' }) {
+function getSituation(contact, selfId) {
```

Comment:
This could be divided into two helpers: `isConfirmed()` and `isFromMe()` for better code clarity, or simply accessing `confirmed` directly and having a helper for just `isFromMe()`

Then you'd use it as so:

```js
const { confirmed } = contact;
if(confirmed && isFromMe(selfId)) {
 // ...
}
```

It just feels like that would be more ergonomic and less fuzzy than "situation" nomenclature.

Not a blocker for merging tho. :-)

Reply:
I like clarifying the logic like this. :+1: Thank you @simison!
Incorporated it into this PR and added more fixes in a98e340.